docs/fhir-guide/FHIR 101 - Practical Guide.ipynb

```diff
@@ -0,0 +1,2966 @@
+    "- `fhirVersion` - the FHIR semantic version number is required. It will instruct  anything validating this resource to use R4 (4.0.0) validation logic.\n",
+    "- `differential` or `snapshot` - Every `StructureDefinition` must have either a `differential` - a section of content that specifies differences from the base definition, or a `snapshot` section which specifies the full definition. For model development, you almost always use the `differential` section and not the `snapshot`."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "hidden": true
+   },
+   "source": [
+    "📘Learn more about [StructureDefinition](http://hl7.org/fhir/structuredefinition.html) anatomy\n",
+    "\n",
+    "📘Learn more about [Resource identifiers](http://hl7.org/fhir/resource.html#identification)\n",
+    "\n",
+    "📘Browse [FHIR data models](https://simplifier.net/search?searchtype=Projects) from other projects"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "heading_collapsed": true,
+    "hidden": true
+   },
+   "source": [
+    "### Remove Attributes"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "hidden": true
+   },
+   "source": [
+    "We're going to start with the easiest modification first: removing an attribute from the base model. In fact, in our initial `Patient` `StructureDefinition`, we've already removed a handful of attributes. \n",
+    "\n",
+    "To remove an attribute you simply set its maximum cardinality to 0 in the `StructureDefinition` `differential` section: \n",
+    "\n",
+    "```json\n",
+    "{\n",
+    "    \"id\": \"Patient.name\",\n",
+    "    \"path\": \"Patient.name\",\n",
+    "    \"max\": \"0\"\n",
+    "}\n",
+    "```"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "heading_collapsed": true,
+    "hidden": true
+   },
+   "source": [
+    "### Make an Attribute Required\n",
+    "\n",
+    "This is just as simple as removing an attribute. To make an attribute required you simply set its minimum cardinality to 1 in the `StructureDefinition` `differential` section:\n",
+    "\n",
+    "```json\n",
+    "{\n",
+    "    \"id\": \"Patient.gender\",\n",
+    "    \"path\": \"Patient.gender\",\n",
+    "    \"min\": 1\n",
+    "}\n",
+    "```\n",
+    "\n",
+    "**A Note on \"mustSupport\"**\n",
+    "\n",
+    "You might have seen the keyword `mustSupport` in some StructureDefinitions. Sometimes it can be confused with `min=1` which makes an attribute required. \n",
+    "\n",
+    "The `mustSupport` keyword actually doesn't have anything to do with making an attribute required. Instead, `mustSupport` means that implementations that produce or consume resources SHALL provide \"support\" for the element in some meaningful way. \n",
```

Comment:
```suggestion
    "The `mustSupport` keyword actually doesn't have anything to do with making an attribute required. Instead, `mustSupport` means that implementations that produce or consume resources [SHALL](https://tools.ietf.org/html/rfc2119) provide \"support\" for the element in some meaningful way. \n",
```

Reply:
I copied this straight from the FHIR spec which uses `SHALL` a lot

Reply:
Ah, here it is. https://plainlanguage.gov/guidelines/conversational/shall-and-must/

Reply:
If you decide to keep "SHALL", maybe link it to https://tools.ietf.org/html/rfc2119 ?

Reply:
Actually I think I'm going to link it to FHIR spec's definition of the conformance language key words: https://www.hl7.org/fhir/conformance-rules.html#conflang